src/compiler/checker.ts

```diff
@@ -18947,6 +18947,12 @@ namespace ts {
             return false;
         }
 
+        function containsTruthyCheck(source: Node, target: Node) {

```

Comment:
A comment here that this specifically deals the binary operator precedence. would also be useful. The implementation looks good here, but this function is named something much more general than it actually is and that's going to be confusing in the future.